File: app/src/main/java/com/_basebase/base/ui/view/FillStatusBarAndNavView.java
package com._basebase.base.ui.view;

import android.content.Context;
import android.os.Build;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.Nullable;

/**
 * 填充 顶部状态栏 和 系统底部导航栏或者叫底部虚拟按钮(返回键和home键和返回首页键)
 */
public class FillStatusBarAndNavView extends View {
    public FillStatusBarAndNavView(Context context) {
        super(context);
    }

    public FillStatusBarAndNavView(Context context, @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public FillStatusBarAndNavView(Context context, @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public FillStatusBarAndNavView(Context context, @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    protected void onAttachedToWindow() {
        super.onAttachedToWindow();
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT_WATCH) {
            requestApplyInsets();
        }
    }
}

File: app/src/main/java/com/_basebase/base/ui/view/FillStatusBarAndNavView.kt
package com._basebase.base.ui.view

import android.content.Context
import android.util.AttributeSet
import android.view.View

/**
 * 填充 顶部状态栏 和 系统底部导航栏或者叫底部虚拟按钮(返回键和home键和返回首页键)
 */
 class FillStatusBarAndNavView : View {
    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        apply {
            requestApplyInsets()
        }
    }
}